instrumentation/kafka-streams/src/main/java/brave/kafka/streams/KafkaStreamsTracing.java

```diff
@@ -1,23 +1,105 @@
+import org.apache.kafka.streams.processor.Processor;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.processor.ProcessorSupplier;
 
-/** Use this class to decorate your Kafka Streams client. */
+/** Use this class to decorate Kafka Stream Topologies and enable Tracing. */
 public final class KafkaStreamsTracing {
 
-  public static KafkaStreamsTracing create(KafkaTracing kafkaTracing) {
-    return new KafkaStreamsTracing(kafkaTracing);
+  final Tracing tracing;
+  final TraceContext.Extractor<Headers> extractor;
+
+  KafkaStreamsTracing(Builder builder) { // intentionally hidden constructor
+    this.tracing = builder.tracing;
+    this.extractor = tracing.propagation().extractor(KafkaStreamsPropagation.GETTER);
+  }
+
+  public static KafkaStreamsTracing create(Tracing tracing) {
+    return new KafkaStreamsTracing.Builder(tracing).build();
+  }
+
+  static void addTags(ProcessorContext processorContext, SpanCustomizer result) {
+    result.tag(KafkaStreamsTags.KAFKA_STREAMS_APPLICATION_ID_TAG, processorContext.applicationId());
+    result.tag(KafkaStreamsTags.KAFKA_STREAMS_TASK_ID_TAG, processorContext.taskId().toString());
+  }
+
+  /**
+   * Creates a {@link KafkaStreams} instance with a tracing-enabled {@link KafkaClientSupplier}. All
+   * Topology Sources and Sinks (including internal Topics) will create a Spans on records
+   * processed.
+   */
+  public KafkaStreams kafkaStreams(Topology topology, Properties streamsConfig) {
+    return new KafkaStreams(topology, streamsConfig, kafkaClientSupplier());
+  }
+
+  /**
+   * Create a tracing-decorated {@link ProcessorSupplier}
+   */
+  public <K, V> ProcessorSupplier<K, V> processor(String name, Processor<K, V> processor) {
+    return new TracingProcessorSupplier<>(this, name, processor);
   }
 
-  final KafkaTracing kafkaTracing;
+  /**
+   * Create a tracing-decorated {@link TransformerSupplier}
+   */
+  public <K, V, R> TransformerSupplier<K, V, R> transformer(String name,
+      Transformer<K, V, R> transformer) {
+    return new TracingTransformerSupplier<>(this, name, transformer);
+  }
+
+  public <V, VR> ValueTransformerSupplier<V, VR> valueTransformer(String name,
+      ValueTransformer<V, VR> valueTransformer) {
+    return new TracingValueTransformerSupplier<>(this, name, valueTransformer);
+  }
 
-  KafkaStreamsTracing(KafkaTracing kafkaTracing) { // intentionally hidden constructor
-    this.kafkaTracing = kafkaTracing;
+  public <K, V, VR> ValueTransformerWithKeySupplier<K, V, VR> valueTransformerWithKey(String name,
+      ValueTransformerWithKey<K, V, VR> valueTransformerWithKey) {
+    return new TracingValueTransformerWithKeySupplier<>(this, name, valueTransformerWithKey);
   }
 
   /** Returns a client supplier which traces send and receive operations. */
-  public KafkaClientSupplier kafkaClientSupplier() {
+  KafkaClientSupplier kafkaClientSupplier() {
```

Comment:
intentionally package private? mention why we need an accessor like this if package private

Reply:
ClientSupplier is only used by `kafkaStreams()` method. I will move this logic to that method.